src/otg/datasource/eqtl_catalogue/study_index.py

```diff
@@ -0,0 +1,151 @@
+    """Study index dataset from eQTL Catalogue."""
+
+    @staticmethod
+    def _all_attributes() -> List[Column]:
+        """A helper function to return all study index attribute expressions.
+
+        Returns:
+            List[Column]: all study index attribute expressions.
+        """
+        study_attributes = [
+            # Project ID, example: "GTEx_V8".
+            f.col("study").alias("projectId"),
+            # Partial study ID, example: "GTEx_V8_Adipose_Subcutaneous". This ID will be converted to final only when
+            # summary statistics are parsed, because it must also include a gene ID.
+            f.concat(f.col("study"), f.lit("_"), f.col("qtl_group")).alias("studyId"),
+            # Summary stats location.
+            f.col("ftp_path").alias("summarystatsLocation"),
+            # Constant value fields.
+            f.lit(True).alias("hasSumstats"),
+            f.lit("eqtl").alias("studyType"),
+        ]
+        tissue_attributes = [
+            # Human readable tissue label, example: "Adipose - Subcutaneous".
+            f.col("tissue_label").alias("traitFromSource"),
+            # Ontology identifier for the tissue, for example: "UBERON:0001157".
+            f.array(
+                f.regexp_replace(
+                    f.regexp_replace(
+                        f.col("tissue_ontology_id"),
+                        "UBER_",
+                        "UBERON_",
+                    ),
+                    "_",
+                    ":",
+                )
+            ).alias("traitFromSourceMappedIds"),
+        ]
+        sample_attributes = [
+            f.lit(838).cast("long").alias("nSamples"),
+            f.lit("838 (281 females and 557 males)").alias("initialSampleSize"),
+            f.array(
+                f.struct(
+                    f.lit(715).cast("long").alias("sampleSize"),
+                    f.lit("European American").alias("ancestry"),
+                ),
+                f.struct(
+                    f.lit(103).cast("long").alias("sampleSize"),
+                    f.lit("African American").alias("ancestry"),
+                ),
+                f.struct(
+                    f.lit(12).cast("long").alias("sampleSize"),
+                    f.lit("Asian American").alias("ancestry"),
+                ),
+                f.struct(
+                    f.lit(16).cast("long").alias("sampleSize"),
+                    f.lit("Hispanic or Latino").alias("ancestry"),
+                ),
+            ).alias("discoverySamples"),
+        ]
+        publication_attributes = [
+            f.lit("32913098").alias("pubmedId"),
+            f.lit(
+                "The GTEx Consortium atlas of genetic regulatory effects across human tissues"
+            ).alias("publicationTitle"),
+            f.lit("GTEx Consortium").alias("publicationFirstAuthor"),
+            f.lit("2020-09-11").alias("publicationDate"),
+            f.lit("Science").alias("publicationJournal"),
+        ]
```

Comment:
That's a good question!

At the moment, the only data in the eQTL Catalogue which is fully ingested and normalised is the GTEx data.

Once they either (a) include other data in the "imported and normalised" section, or (b) we decide to run the harmonisation on our side, we'll certainly have to face this question.

We certainly should keep this in mind for the future, but I think for the current iteration we can keep things as is. (But please let me know if you think otherwise!)

Reply:
Oh, I thought we chose GTEx from the many studies within the eQTL catalog.

Reply:
That's right, but among all of the studies, GTEx was the only one fully harmonised, so we didn't have much choise :upside_down_face: